packages/terra-popup/tests/wdio/popup-spec.js

```diff
@@ -1,499 +1,360 @@
-    describe('Bounded Popup viewports', () => {
-      before(() => {
-        browser.url('/#/raw/tests/terra-popup/popup/bounded-popup-closed');
-        browser.setViewportSize(viewport);
-        browser.click('#bounded-button');
-      });
-
-      Terra.it.matchesScreenshot({ selector: '#test-popup-area' });
-    });
-  });
-
-  describe('Displays a default popup & closes on width resize', () => {
-    before(() => browser.setViewportSize(Terra.viewports('medium')[0]));
-    beforeEach(() => {
+Terra.describeViewports('Popup', ['medium'], () => {
+  describe('Default popup', () => {
+    it('validates popup does not have an arrow', () => {
       browser.url('/#/raw/tests/terra-popup/popup/default-popup');
       browser.waitForVisible('.test-content');
-      browser.setViewportSize(Terra.viewports('small')[0]);
+      expect(browser.isExisting('.test-arrow')).to.be.false;
+      Terra.validates.screenshot({ selector });
     });
 
-    Terra.it.matchesScreenshot({ selector: '#test-popup-area' });
-  });
-
-  describe('Displays a default popup & remains open on height resize', () => {
-    before(() => browser.setViewportSize(Terra.viewports('medium')[0]));
-    beforeEach(() => {
-      browser.url('/#/raw/tests/terra-popup/popup/default-popup');
-      browser.click('#default-button');
-      browser.waitForVisible('.test-content');
-      browser.setViewportSize({ width: browser.getViewportSize('width'), height: 600 });
+    it('closes on ESC', () => {
+      browser.keys('Escape');
+      expect(browser.isExisting('.test-content')).to.be.false;
     });
 
-    Terra.it.matchesScreenshot({ selector: '#test-popup-area' });
-  });
-
-  describe('Displays a default popup & closes on ESC', () => {
-    before(() => browser.setViewportSize(Terra.viewports('medium')[0]));
-    beforeEach(() => {
-      browser.url('/#/raw/tests/terra-popup/popup/default-popup');
+    it('reopens the popup', () => {
       browser.click('#default-button');
       browser.waitForVisible('.test-content');
-      browser.keys('ESCAPE');
     });
 
-    Terra.it.matchesScreenshot({ selector: '#test-popup-area' });
-  });
+    it('closes on outside click', () => {
+      browser.moveToObject('#root', 300, 300).leftClick();
+      expect(browser.isExisting('.test-content')).to.be.false;
+    });
 
-  describe('Displays a default popup & closes on outside click', () => {
-    before(() => browser.setViewportSize(Terra.viewports('medium')[0]));
-    beforeEach(() => {
-      browser.url('/#/raw/tests/terra-popup/popup/default-popup');
+    it('reopens the popup', () => {
       browser.click('#default-button');
       browser.waitForVisible('.test-content');
-      browser
-        .moveToObject('#root', 300, 300)
-        .leftClick();
     });
 
-    Terra.it.matchesScreenshot({ selector: '#test-popup-area' });
-  });
+    it('remains open on height resize', () => {
+      browser.setViewportSize({ width: mediumViewport.width, height: 600 });
```

Comment:
I like to grab the current viewport width, then this would also work in the loops
```javascript
const viewport = browser.getViewportSize();
viewport.height = 600;
browser.setViewportSize(viewport);
```

Reply:
@mjhenkes Good point. Since we only tested one viewport I didn't think about this impact

Reply:
resolved: https://github.com/cerner/terra-framework/pull/734/commits/0afcefd17e8a4c5e3f0b94be3d8fd1ad28cbd347#diff-75a22fd845592219495fcc77ea6104c3R6